portfolio/src/main/webapp/scripts/servlet.js

```diff
@@ -0,0 +1,106 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Calls to Datastore servlet to load and add comments
+ */
+
+
+/** Takes the max number of comments available to the user from datastore */
+function setMaxNumberOfComments() {
+  fetch('/comments-size').then(response => response.text()).then((commentArraySize)  => {
+    const maxArraySize = parseInt(commentArraySize);
+    const maxNumberofComments = document.getElementById('number-of-comments');
+
+    maxNumberofComments.max = maxArraySize.toString(); // Set the max number as attribute of the input
+  });
+}
+
+/** Takes the number of comments the user wants to see and displays them */
+function loadComments() {
+  const numberOfComments = document.getElementById('number-of-comments').value;
+  fetch(`/load-comments?number-of-comments=${numberOfComments}`)
+    .then(response => {
+      if(response.redirected) {
+        alert('Login to load the comments');
+        return;
+      } else {
+        response.json().then((comments) => {
+          setMaxNumberOfComments(); // Update the maximum number of comments available.
+          removeCommentsFromDOM(); // Remove passed comments before loading new ones
+          addCommentsToDOM(comments);
+        });
+      }
+  });
+}
+
+/** Eliminate displayed comments to load a new set of them */
+function removeCommentsFromDOM() {
+  const commentWrapper = document.getElementById('comment-display-container');
+
+  while(commentWrapper.firstChild) { 
+    // While there are comments left, remove them
+    commentWrapper.removeChild(commentWrapper.firstChild);
+  }
+}
+
+/** Display the number selected of comments */
+function addCommentsToDOM(comments) {
+  const commentWrapper = document.getElementById('comment-display-container');
+  
+  for (let index = 0; index < comments.length; index++) {
+    const userComment = comments[index];
+
+    /** Defined template of the comment card */
+    const userCommentTemplate = document.getElementsByTagName('template')[0];
+    const templateClone = userCommentTemplate.content.cloneNode(true);
+
+    /** Add the information to the card */
+    templateClone.querySelector('b').textContent = userComment.user;
+    templateClone.querySelector('p').textContent = userComment.comment;
+
+    /** If the remove buttons is clicked, remove the comment */
+    templateClone.getElementById('close-button-wrapper').addEventListener('click', () => {
+      handleDeleteCommentRequest(userComment);
+    });
+
+    /** Add the comment to the comments container */
+    commentWrapper.appendChild(templateClone); 
+  }
+}
+
+/** Checks if the user is available to delete the comment  */
+function handleDeleteCommentRequest(userComment) {
```

Comment:
```suggestion
/** @param {{user: string, comment: string}} */
function handleDeleteCommentRequest(userComment) {
```
Remember to use JSDoc on functions if they have parameters or return values :)

Reply:
Thank you Aki!